File: uptmdigital_app/lib/models/inscripcion.dart
class Inscripcion {
  final int idInscripcion;
  final int estudianteId;
  final int asignaturaId;
  final int? periodoId;
  final String estado;
  final String periodoNombre;

  Inscripcion({
    required this.idInscripcion,
    required this.estudianteId,
    required this.asignaturaId,
    this.periodoId,
    required this.estado,
    this.periodoNombre = '',
  });

  factory Inscripcion.fromJson(Map<String, dynamic> json) {
    String per = '';
    if (json['periodo'] is Map) {
      per = json['periodo']['nombre']?.toString() ?? '';
    } else if (json['periodoNombre'] != null) {
      per = json['periodoNombre'].toString();
    } else if (json['periodo'] != null) {
      per = json['periodo'].toString();
    }

    return Inscripcion(
      idInscripcion: json['idInscripcion'] ?? 0,
      estudianteId: json['estudianteId'] ?? 0,
      asignaturaId: json['asignaturaId'] ?? 0,
      periodoId: json['periodoId'],
      estado: json['estado'] ?? '',
      periodoNombre: per,
    );
  }
}

File: UPTMDigital.API/Models/Inscripcion.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace UPTMDigital.API.Models
{
    [Table("Inscripcion")]
    public class Inscripcion
    {
        [Key]
        public int IdInscripcion { get; set; }

        // --- FK Normalizadas ---
        public int EstudianteId { get; set; }
        [ForeignKey("EstudianteId")]
        public Estudiante Estudiante { get; set; } = null!;

        public int AsignaturaId { get; set; }
        [ForeignKey("AsignaturaId")]
        public Asignatura Asignatura { get; set; } = null!;

        public int? PeriodoId { get; set; }
        [ForeignKey("PeriodoId")]
        public Periodo? Periodo { get; set; }

        public DateTime? FechaInscripcion { get; set; }
        public string? Estado { get; set; }
        public bool Activo { get; set; } = true;
    }
}